### PR TITLE
rbac: remove unwaned rbac for the leader election

### PR DESCRIPTION
**Pull request by humblec**
Considering we have removed leader election based on endpoints.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

> /kind cleanup

-->
```release-note
NONE
```

### CODE REVIEW

File: deploy/kubernetes/rbac.yaml [hunk 1]
```diff
@@ -26,37 +26,37 @@ rules:
...
-    resources: ["volumesnapshotcontents"]
-    verbs: ["get", "list"]
-  - apiGroups: ["storage.k8s.io"]
-    resources: ["csinodes"]
-    verbs: ["get", "list", "watch"]
-  - apiGroups: [""]
-    resources: ["nodes"]
-    verbs: ["get", "list", "watch"]
+  - apiGroups: [ "" ]
+    resources: [ "persistentvolumes" ]
```
Comment (pohly) [inline deploy/kubernetes/rbac.yaml:30]: Can you remove the unnecessary whitespace changes?

Comment (humblec) [reply deploy/kubernetes/rbac.yaml:30]: sure.. @pohly

Comment (humblec) [reply deploy/kubernetes/rbac.yaml]: @pohly done.. thanks

File: deploy/kubernetes/rbac.yaml [hunk 2]
```diff
@@ -83,14 +83,6 @@ metadata:
...
   name: external-provisioner-cfg
 rules:
-# Only one of the following rules for endpoints or leases is required based on
-# what is set for `--leader-election-type`. Endpoints are deprecated in favor of Leases.
-- apiGroups: [""]
-  resources: ["endpoints"]
-  verbs: ["get", "watch", "list", "delete", "update", "create"]
-- apiGroups: ["coordination.k8s.io"]
-  resources: ["leases"]
-  verbs: ["get", "watch", "list", "delete", "update", "create"]
```
Comment (pohly) [inline deploy/kubernetes/rbac.yaml:93]: The PR description only talks about "Considering we have removed leader election based on endpoints". Then why are the RBAC rules for leases also getting removed?

Comment (humblec) [reply deploy/kubernetes/rbac.yaml:93]: adjusted .. thanks.

File: deploy/kubernetes/rbac.yaml [hunk 3]
```diff
@@ -85,9 +85,6 @@ metadata:
 rules:
 # Only one of the following rules for endpoints or leases is required based on
 # what is set for `--leader-election-type`. Endpoints are deprecated in favor of Leases.
-- apiGroups: [""]
-  resources: ["endpoints"]
-  verbs: ["get", "watch", "list", "delete", "update", "create"]
```
Comment (pohly) [inline deploy/kubernetes/rbac.yaml:90]: This is the same change that was made earlier in https://github.com/kubernetes-csi/external-resizer/commit/15a912cfe617c0d38a84cd2b8639c2013b912358

/lgtm
